### ISSUE TITLE
Warnings introduced by gcc10 + C++17

### ISSUE THREAD

**Issue by yipengsun** [umd-lhcb/TrackerOnlyEmu#4]
We now use ROOT that is compiled w/ C++ 17 standard. Now I see the following warnings when running the `scripts/run2-rdx-l0_hadron.py`:

```
input_line_75:10:7: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
      ((const vector<double>*)obj)->empty();
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
input_line_137:10:7: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
      ((const vector<unsigned int>*)obj)->empty();
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
input_line_144:10:7: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
      ((const vector<ULong64_t>*)obj)->empty();
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

It turns out that these warnings are caused by the following lines:
https://github.com/umd-lhcb/TrackerOnlyEmu/blob/e9d81096551544ddbed6c78216701fec519c518f/scripts/run2-rdx-l0_hadron.py#L178-L179

https://github.com/umd-lhcb/TrackerOnlyEmu/blob/e9d81096551544ddbed6c78216701fec519c518f/scripts/run2-rdx-l0_hadron.py#L187

Basically, the TTree -> NumPy conversion implemented in ROOT is not full C++ 17 compliant.

**Comment by yipengsun** [comment]
Looks like a known problem to ROOT developers: https://github.com/root-project/root/issues/8622.

**Comment by yipengsun** [comment]
This is resolved in ROOT 6.26/00.
